accounts-db/src/io_uring/memory.rs

```diff
@@ -120,49 +130,148 @@ impl DerefMut for PageAlignedMemory {
     }
 }
 
-/// Fixed mutable view into externally allocated bytes buffer
+/// Fixed mutable view into externally allocated IO bytes buffer
+/// registered in `io_uring` for access in scheduled IO operations.
 ///
-/// It is an unsafe (no lifetime tracking) equivalent of `&mut [u8]`
-pub struct BorrowedBytesMut {
+/// It is used as an unsafe (no lifetime tracking) equivalent of `&mut [u8]`.
+pub(super) struct IoFixedBuffer {
     ptr: *mut u8,
     size: usize,
+    io_buf_index: Option<u16>,
 }
 
-impl BorrowedBytesMut {
+impl IoFixedBuffer {
     pub const fn empty() -> Self {
         Self {
             ptr: std::ptr::null_mut(),
             size: 0,
+            io_buf_index: None,
         }
     }
 
-    pub fn from_mut_slice(buf: &mut [u8]) -> Self {
-        Self {
-            ptr: buf.as_mut_ptr(),
-            size: buf.len(),
-        }
+    /// Split buffer into `chunk_size` sized `IoFixedBuffer` buffers for use as registered
+    /// buffer in io_uring operations.
+    pub fn split_buffer_chunks<'a>(
```

Comment:
I don't think that this code should be here, because it makes it impossible to
track the lifetime of the memory.

I would move this chunking/registering to what actually owns the buffer, so it's
clear from there that even thought we're downgrading to pointers, they won't be dangling

Reply:
This function is actually just a factory of IoFixedBuffer items and the caller, which is the owner of the input buffer too, manages the result - the code here doesn't leak the unsafe pointers anywhere else.

Also, the code was moved here because it's shared between file creator and file sequential reader.

Reply:
> This function is actually just a factory of IoFixedBuffer items and the caller, which is the owner of the input buffer too, manages the result 

This is exactly the problem tho: the code returns values that embed pointers, but it's up to the caller to guarantee that the backing memory for those pointers remain valid. 

Or in other words: this is a safe API, that can be misused to trigger use after free. Safe APIs should never allow use after free.

Reply:
I suppose for that reason it will be best to mark those functions as unsafe. I prefer to keep the constructor with the struct being constructed (also copy-pasting unsafe code to each use place seem quite counter-productive), but indeed this operation is unsafe.